### PR TITLE
doc: Add guide for migrating to config and disk resources.

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

The guide to explain how to migrate from embedded configs and disks in `linode_instance` resource to standalone `linode_instance_config` and `linode_instance_disk` resources.

### CODE REVIEW

File: docs/guides/migrate_to_instance_config_and_disk_resources.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+attributes in `linode_instance` resource to be `linode_instance_config`
+and `linode_instance_disk` resource.
+
+-> An experimental feature, generating configuration, will be used in this guide.
+For details, please check
+[Terraform document site](https://developer.hashicorp.com/terraform/language/import/generating-configuration).
+
+## Create a simple instance with embedded config and disk
+
+This is a sample TF config file for a linode instance containing deprecated
```
Comment (lgarber-akamai) [inline docs/guides/migrate_to_instance_config_and_disk_resources.md:13]:
```suggestion
This is a sample TF config file for a Linode instance containing deprecated
```

File: docs/guides/migrate_to_instance_config_and_disk_resources.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+```
+
+## Future
+
+Ideally, we would like to import the resources with ids stored in the states,
+but it's currently not supported by the TF.
+
+I put up an [issue](https://github.com/hashicorp/terraform/issues/33880)
+in their repo for it, and later realized they already implemented it. This feature will be available in TF v1.6
+according to their plan on GitHub.
```
Comment (lgarber-akamai) [inline docs/guides/migrate_to_instance_config_and_disk_resources.md:227]: I think we can drop the issue reference and just say this feature will be available in `v1.6` 👍

**Review by lgarber-akamai — APPROVED** [review]
Just a few small wording adjustments, otherwise this looks good!

**Comment by zliang-akamai** [comment]
@lgarber-akamai Thanks for the review! I applied your suggestions alongside with some other suggestions from chatGPT.

**Review by lgarber-akamai — APPROVED** [review]

**Review by yec-akamai — APPROVED** [review]

**Pull request merged**